File: thetvdb/initialBuildTvDb.pl
#!/usr/bin/perl
# This file is part of thetvdb.
#
# thetvdb is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# thetvdb is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with thetvdb. If not, see <http://www.gnu.org/licenses/>.

use strict;

use Getopt::Long;
use Data::Dumper;
use lib ".";
use parseTvDbXml;

my $apiKey;
my $seriesId;
my $verbose;
my $header;
my $help;

GetOptions("apiKey=s" => \$apiKey,
           "seriesId=i" => \$seriesId,
           "verbose" => \$verbose,
           "header" => \$header,
           "help|?|h" => \$help) or die("Error in command line arguments\n");

help() if ($help or !$apiKey or !$seriesId);

my $parser = new parseTvDbXml();

print "Retrieve server time ... " if ($verbose);
my $serverTime = $parser->getServerTime();
print $serverTime . "\n";

print "Retrieve serie ... \n" if ($verbose);
print "seriesId|seasonId|episodeId|seasonNr|episodeNr|airdate|episodeName\n" if ($header);
my @serie_list = $parser->retrieveSerie($seriesId, $apiKey);

binmode(STDOUT, ":utf8"); # prevent warning: "Wide character in print"
for my $episode_list_ref (@serie_list) {
    print join('|', @{$episode_list_ref}) . "\n";
}


#print Dumper(@serie_list);

print "finished.\n" if ($verbose);
exit;

sub help {
  print <<HELP

Script downloads information about a given serie from http://www.thetvdb.com.
The downloaded ZIP file is extracted and the XML file parsed in order to
retrieve the episodes from the serie.

$0 [options]

options:
  -apiKey <tvDbAPIKey>      API key for accessing thetvdb.com content
  -seriesId <seriesId>      Id of the serie as integer
  -header                   (optional) If set, column header is printed
  -verbose                  (optional) If set, script prints more information
                            during runtime
  -h|help|?                 (optional) This page

HELP
;
  exit 0;
}

File: thetvdb/initialBuildTvDb.php
<?php
/**
* This file is part of thetvdb.
*
* thetvdb is free software: you can redistribute it and/or modify
* it under the terms of the GNU General Public License as published by
* the Free Software Foundation, either version 3 of the License, or
* (at your option) any later version.
*
* thetvdb is distributed in the hope that it will be useful,
* but WITHOUT ANY WARRANTY; without even the implied warranty of
* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
* GNU General Public License for more details.
*
* You should have received a copy of the GNU General Public License
* along with thetvdb. If not, see <http://www.gnu.org/licenses/>.
* -----------------------------------------------------------------------------
* Script downloads information about a given serie from http://www.thetvdb.com.
* The downloaded ZIP file is extracted and the XML file parsed in order to
* retrieve the episodes from the serie.
**/
include_once("tvDbParser.php");

$shortopts = "h::a:s:";

$longopts = array(
  "help::",
  "apiKey:",
  "seriesId:"
);

$options = getopt($shortopts, $longopts);

if (array_key_exists("help", $options) || array_key_exists("h", $options) ||
    !(array_key_exists("apiKey", $options) || array_key_exists("a", $options)) ||
    !(array_key_exists("seriesId", $options) || array_key_exists("s", $options))) {
  help();
}

$apiKey = (array_key_exists("apiKey", $options) ? $options['apiKey'] : $options['a']);
$seriesId = (array_key_exists("seriesId", $options) ? $options['seriesId'] : $options['s']);

$ret = retrieveSeriesById($apiKey, $seriesId);
if ($ret) {
  $episodes = parseEpisodeXml('en.xml');
  foreach ($episodes as $episode) {
    print sprintf("INSERT INTO tv2_episode (id, tvdb_seriesId, firstAired, season, episode, episode_title) VALUES (%s, %s, '%s', %s, %s, '%s');\n", 
      $episode['id'],
      $episode['seriesid'],
      $episode['FirstAired'],
      $episode['SeasonNumber'],
      $episode['EpisodeNumber'],
      addslashes($episode['EpisodeName']));
  }
}

return 0;

function help() {
  $filename = basename(__FILE__);
  print <<<EOT

Script downloads information about a given serie from http://www.thetvdb.com.
The downloaded ZIP file is extracted and the XML file parsed in order to
retrieve the episodes from the serie.

$filename [options]

options:
  -a <tvDbAPIKey>             API key for accessing thetvdb.com content
  --apiKey <tvDbAPIKey>
  -s <seriesId>               Id of the serie as integer
  --seriesId <seriesId>
  -h                          (optional) This page
  --help

EOT;

  exit;
}

?>